llama_stack/providers/utils/datasetio/dataset_utils.py

```diff
@@ -0,0 +1,31 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from llama_stack.apis.datasetio import *  # noqa: F403
+
+
+class BaseDataset(ABC):
```

Comment:
So I actually think we should keep BaseDataset definition still within meta-reference. We should not share "base classes" -- what we should share are "utilities". The design is that we make APIs and providers provide implementations. They can design their internals however they decide having inherited classes or not. We can offer utilities which should take the form of pure functions ideally and "mix-ins" at best.

Reply:
SG, killing dataset_utils.py.